common/eventconsumption/eventindexmodelstorage.cpp

```diff
@@ -101,8 +109,31 @@ void Storage::configureFiles(const IPropertyTree& config)
     ForEach(*it)
     {
         File file;
-        file.path.set(it->query().queryProp("@path"));
-        file.plane = &lookupPlane(it->query().queryProp("@plane"), file.path.str());
+        const IPropertyTree& fileConfig = it->query();
+        file.path.set(fileConfig.queryProp("@path"));
+        // allow the configuration to place the entire file in a single plane
+        if (fileConfig.hasProp("@plane"))
+        {
+            const Plane& plane = lookupPlane(fileConfig.queryProp("@plane"), file.path.str());
+            for (size_t i = 0; i < File::NumPlanes; ++i)
+                file.planes[i] = &plane;
+        }
+        // branch nodes may be stored separately from leaf nodes
+        if (fileConfig.hasProp("@branchPlane"))
+        {
+            const Plane& plane = lookupPlane(fileConfig.queryProp("@branchPlane"), file.path.str());
+            file.planes[0] = &plane;
+        }
+        else if (!file.planes[0])
+            file.planes[0] = defaultPlane;
+        // leaf nodes may be stored separately from branch nodes
+        if (fileConfig.hasProp("@leafPlane"))
+        {
+            const Plane& plane = lookupPlane(fileConfig.queryProp("@leafPlane"), file.path.str());
+            file.planes[1] = &plane;
+        }
+        else if (!file.planes[1])
+            file.planes[1] = defaultPlane;
         if (!configuredFiles.insert(file).second)
```

Comment:
insert(std::move(file)) could be used - but you would need to use a different variable instead of file.path (e.g. assign to a variable at the head).